Broken tests on PowerPC64 (Little Endian) and System Z architecturs

119 failed tests for PPC64LE - https://travis-ci.com/radareorg/radare2/jobs/256856419

See also build warnings:
```
p/debug_native.c: In function ‘r_debug_native_bp’:
p/debug_native.c:1421:10: warning: unused variable ‘dbg’ [-Wunused-variable]
  RDebug *dbg = bp->user;
          ^~~
p/native/linux/linux_debug.c: In function ‘print_fpu’:
p/native/linux/linux_debug.c:1058:2: warning: #warning print_fpu not implemented for this platform [-Wcpp]
 #warning print_fpu not implemented for this platform
  ^~~~~~~
p/native/linux/linux_debug.c: In function ‘linux_reg_read’:
p/native/linux/linux_debug.c:1063:7: warning: variable ‘showfpu’ set but not used [-Wunused-but-set-variable]
  bool showfpu = false;
       ^~~~~~~
p/native/linux/linux_debug.c: In function ‘linux_reg_write’:
p/native/linux/linux_debug.c:1268:55: warning: passing argument 4 of ‘r_debug_ptrace’ discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
   int ret = r_debug_ptrace (dbg, PTRACE_SETREGS, pid, buf, NULL);
                                                       ^~~
In file included from p/native/linux/linux_debug.c:6:0:
/home/travis/build/radareorg/radare2/libr/include/r_debug.h:608:20: note: expected ‘void *’ but argument is of type ‘const unsigned char *’
 static inline long r_debug_ptrace(RDebug *dbg, r_ptrace_request_t request, pid_t pid, void *addr, r_ptrace_data_t data) {
                    ^~~~~~~~~~~~~~
At top level:
p/native/linux/linux_debug.c:970:13: warning: ‘print_fpu’ defined but not used [-Wunused-function]
 static void print_fpu(void *f){
             ^~~~~~~~~
+make install
```

--- a/libr/egg/egg.c
+++ b/libr/egg/egg.c
@@ -316,6 +316,14 @@ R_API void r_egg_printf(REgg *egg, const char *fmt, ...) {
 }
 
 R_API bool r_egg_assemble_asm(REgg *egg, char **asm_list) {
+	// Nothing to translate: -P/-B/-w style invocations produce raw bytes
+	// without going through the egg compiler, so they shouldn't require a
+	// backend for the configured arch.
+	char *code = r_buf_tostring (egg->buf);
+	if (R_STR_ISEMPTY (code)) {
+		free (code);
+		return true;
+	}
 	char *asm_name = NULL;
 	if (asm_list) {
 		char **asm_ = asm_list;
@@ -343,27 +351,18 @@ R_API bool r_egg_assemble_asm(REgg *egg, char **asm_list) {
 		r_asm_set_big_endian (egg->rasm, egg->endian);
 		// r_asm_set_syntax (egg->rasm, R_ARCH_SYNTAX_INTEL);
 		r_arch_config_set_syntax (egg->rasm->config, R_ARCH_SYNTAX_INTEL);
-		char *code = r_buf_tostring (egg->buf);
-		if (R_STR_ISEMPTY (code)) {
-			free (code);
-			if (r_buf_size (egg->bin) == 0) {
-				R_LOG_DEBUG ("The egg compiler generated no code to assemble");
-			}
+		RAsmCode *asmcode = r_asm_assemble (egg->rasm, code);
+		if (asmcode && asmcode->len > 0) {
 			ret = true;
+			r_buf_append_bytes (egg->bin, asmcode->bytes, asmcode->len);
 		} else {
-			RAsmCode *asmcode = r_asm_assemble (egg->rasm, code);
-			if (asmcode && asmcode->len > 0) {
-				ret = true;
-				r_buf_append_bytes (egg->bin, asmcode->bytes, asmcode->len);
-			} else {
-				R_LOG_ERROR ("r_asm_assemble has failed %s", code);
-			}
-			r_asm_code_free (asmcode);
-			free (code);
+			R_LOG_ERROR ("r_asm_assemble has failed %s", code);
 		}
+		r_asm_code_free (asmcode);
 	} else {
 		R_LOG_ERROR ("Cannot find a valid assembler");
 	}
+	free (code);
 	return ret;
 }
 

--- a/libr/egg/egg.c
+++ b/libr/egg/egg.c
@@ -316,9 +316,6 @@ R_API void r_egg_printf(REgg *egg, const char *fmt, ...) {
 }
 
 R_API bool r_egg_assemble_asm(REgg *egg, char **asm_list) {
-	// Nothing to translate: -P/-B/-w style invocations produce raw bytes
-	// without going through the egg compiler, so they shouldn't require a
-	// backend for the configured arch.
 	char *code = r_buf_tostring (egg->buf);
 	if (R_STR_ISEMPTY (code)) {
 		free (code);

--- a/test/db/tools/ragg2
+++ b/test/db/tools/ragg2
@@ -371,6 +371,14 @@ AA
 EOF
 RUN
 
+NAME=ragg2 debruijn sequence on arch without egg backend
+FILE=-
+CMDS=!ragg2 -a ppc -b 64 -r -P 2; ?e
+EXPECT=<<EOF
+AA
+EOF
+RUN
+
 NAME=ragg2 -H R2_NOPLUGINS
 FILE=-
 CMDS=!R2_NOPLUGINS=1;ragg2 -H R2_NOPLUGINS

